utils/service_discovery/sd_docker_backend.py

```diff
@@ -223,6 +230,14 @@ def _get_additional_tags(self, container_inspect, *args):
             tags.append('pod_name:%s' % pod_metadata.get('name'))
         return tags
 
+    def _get_kube_container_name(self, c_id):
+        pods = self.kubeutil.retrieve_pods_list().get('items', [])
```

Comment:
not super excited about calling this for every container, can we maybe build the mapping `c_id -> c_name` once before entering this loop https://github.com/mikekap/dd-agent/blob/4790dd8df8417900a62fb776f0e10fa3491fb749/utils/service_discovery/sd_docker_backend.py#L258 to reduce api calls? It should be fine since it's coming from the kubelet but with nodes running hundreds of containers if there's a hiccups we'll feel it.